clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -11227,6 +11227,8 @@ static llvm::Type *getSPIRVImageType(llvm::LLVMContext &Ctx, StringRef BaseType,
 llvm::Type *CommonSPIRTargetCodeGenInfo::getOpenCLType(CodeGenModule &CGM,
                                                        const Type *Ty) const {
   llvm::LLVMContext &Ctx = CGM.getLLVMContext();
+  if (Ctx.supportsTypedPointers())
```

Comment:
Does it make sense to do that only for SYCL?

Reply:
Running a lot of the OpenCL tests, not doing this for OpenCL causes far, far more issues.

Reply:
Is that because community doesn't support typed pointers anymore?

Reply:
More that the OpenCL CPU backend doesn't really support opaque pointers yet, nor the target extension types.